RFCs/rfc-0008-wallet-provider-api.md

```diff
@@ -0,0 +1,387 @@
+
+#### Handling Adversarial Behavior
+
+Since it is a JavaScript object, consumers can generally perform arbitrary operations on the Provider, and all its properties can be read or overwritten. Therefore, it is best to treat the Provider object as though it is controlled by an adversary. It is paramount that the Provider implementer protects the user, Wallet, and Client by ensuring that:
+
+- The Provider does not contain any private user data.
+- The Provider and Wallet programs are isolated from each other.
+- The Wallet and/or Client rate-limit requests from the Provider.
+- The Wallet and/or Client validate all data sent from the Provider.
+
+### Provider RPC Methods
+This section lists the standard RPC methods that can be supported by the Provider, along with their return types. Each method follows the `mina_{methodName}` naming convention.
+
+#### mina_accounts
+Returns a list of addresses owned by client.
+
+##### Parameters
+None
+
+##### Returns
+An array of hexadecimals as strings representing the addresses owned by the client.
+
+```typescript
+Provider.request({ 
+  method: 'mina_accounts'
+}): Promise<string[]>;
+```
+
+#### mina_chainId
+Returns the chain ID of the current network.
+
+##### Parameters
+None
+
+##### Returns
+A hexadecimal of the current chain ID.
+
+```typescript
+Provider.request({
+    method: 'mina_chainId'
+}): Promise<string>;
+```
+
+#### mina_sign
+Returns a signed message.
+
+##### Parameters
+`message`: [Required] a string
+
+##### Returns
+A signed message.
+
+```typescript
+type SignedMessage = {
+    publicKey: string;
+    data: string;
+    signature: {
+        field: string;
+        scalar: string;
+    };
+};
+
+Provider.request({
+    method: 'mina_sign',
+    params: { message: "hi bob" }
+}): Promise<SignedMessage>;
+```
+
+#### mina_signFields
+Returns a signed message.
+
+##### Parameters
+`fields`: [Required] an array of numbers or strings
+
+##### Returns
+
+```typescript
+type SignedFieldsData = {
+    publicKey: string;
+    data: (string | number)[];
+    signature: {
+        field: string;
+        scalar: string;
+    };
+};
+
+Provider.request({
+    method: 'mina_signFields',
+    params: { fields: [1, 5, 9, 2] }
+}): Promise<SignedFieldsData>;
+```
+
+#### mina_signTransaction
+Returns a signed transaction.
+
+##### Parameters
+`transaction`: [Required] a well-formed transaction body.
+
+##### Returns
+A signed transaction body.
```

Comment:
Why does zkApp need a signed transaction body ?

Reply:
This can also be sign and submit, but not all scenarios may require a transaction to be submitted immediately after being signed.